pyquil/gates.py

```diff
@@ -443,18 +443,18 @@ def PSWAP(angle, q1, q2):
 """
 
 
-def RESET(qubit_index=None):
+def RESET(qubit=None):
```

Comment:
What is the reasoning for changing the kwarg name here? This change will be backwards incompatible.

Reply:
The original form of the function required qubit_index to be an int, which was the index of the qubit. This is fine for assigned qubits, but it prevented me from using QubitPlaceholder. I changed it to use unpack_qubit() instead, which means you no longer need to provide it with an index. You can give it an int, a Qubit, or a QubitPlaceholder. Since you are no longer really providing an index but an actual qubit, it made sense to rename it. This is how all of the other gates work, and they all have `qubit` as the name of their argument. 

If it will break backwards compatibility, you can leave it called qubit_index, but it will be confusing since it doesn't take an index anymore.

Reply:
Backwards compatibility is a must, unfortunately. We have historically been cavalier with BC-breaking changes, but going forward we will be strictly BC. 

If you feel the clarity of the kwarg is a must here, then the (maybe ugly) compromise is to have two kwargs `def RESET(qubit_index=None, qubit=None)`. I can't see this breaking BC since this maintains both the kwarg and its position (so something like `RESET(qi)`, where `qi` is a number, would continue to work). You will need some extra logic like `qubit = qubit_index if qubit_index is not None` or something.

Reply:
If it's a hard rule, then I'll change it back to qubit_index and just update the docstring to clarify that you can pass a qubit in as well. As long as I can reset a QubitPlaceholder, I don't particularly care what the parameter is called.